components/core/src/streaming_archive/writer/Archive.cpp

```diff
@@ -307,26 +324,23 @@ namespace streaming_archive { namespace writer {
      * 1) Given file has a timestamp, so append to segment only if enough files are buffered
      * 2) Given file lacks a timestamp, append to segment immediately
      */
-    void Archive::mark_file_ready_for_segment (File*& file) {
-        // Check if file actually exists in our tracked file container
-        if (m_mutable_files.count(file) == 0) {
-            SPDLOG_CRITICAL("Unable to mark a file ready for segment for file that is not tracked by the current archive");
-            throw OperationFailed(ErrorCode_Unsupported, __FILENAME__, __LINE__);
-        }
+    void Archive::mark_file_ready_for_segment () {
 
-        // Remove file pointer visibility from outside of the archive
-        m_mutable_files.erase(file);
-
-        if (file->has_ts_pattern()) {
-            append_file_to_segment(file, m_segment_for_files_with_timestamps, m_logtype_ids_in_segment_for_files_with_timestamps,
+        if (m_file->has_ts_pattern()) {
```

Comment:
Before this, we should probably check if the file is set and throw an exception if not.